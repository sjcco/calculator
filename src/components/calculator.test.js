import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { render, within } from '@testing-library/react';
import Calculator from './calculator';

test('should render page correctly', () => {
  const calc = renderer.create(<Calculator />).toJSON();
  expect(calc).toMatchSnapshot();
});

test('renders component', () => {
  const component = render(<Calculator />);
  component.getByText("Let's do some math!");
});

test('Renders all buttons in calculator', () => {
  render(<Calculator />);
  const buttons = within(document.querySelector('.button-panel')).getAllByRole('button');
  expect(buttons).toHaveLength(19);
});
