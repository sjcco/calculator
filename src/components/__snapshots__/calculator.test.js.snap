// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render page correctly 1`] = `
<div
  className="calc-container"
>
  <h1>
    Let's do some math!
  </h1>
  <div
    className="calculator skin"
  >
    <div
      className="display skin"
    >
      <p />
      <p>
        0
      </p>
    </div>
    <div
      className="button-panel skin"
    >
      <div
        className="btn-group"
      >
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          AC
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          +/-
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          %
        </button>
        <button
          className="button bg-orange"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          7
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          8
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          9
        </button>
        <button
          className="button bg-orange"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          4
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          5
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          6
        </button>
        <button
          className="button bg-orange"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          1
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          2
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          3
        </button>
        <button
          className="button bg-orange"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
      </div>
      <div
        className="btn-group"
      >
        <button
          className="button bg-gray zero"
          onClick={[Function]}
          type="button"
        >
          0
        </button>
        <button
          className="button bg-gray"
          onClick={[Function]}
          type="button"
        >
          .
        </button>
        <button
          className="button bg-orange"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
